### PR TITLE
Add hardEviction

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Fixes: 8636074

### What this PR does / why we need it:

Adds hardEviction to nodes

### Test plan for the issue:

Locally run busy nodes and see

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

**Comment by mjudeikis** [comment]
We need to be aware of this issue: https://github.com/Azure/ARO-RP/issues/1103 where We can't remove values at this point in time

File: pkg/operator/controllers/workaround/systemreserved.go [hunk 1]
```diff
@@ -60,11 +61,15 @@ func (sr *systemreserved) IsRequired(clusterVersion *version.Version) bool {
 }
 
 func (sr *systemreserved) kubeletConfig() (*unstructured.Unstructured, error) {
-	kubeletConfig := map[string]map[string]string{
```
Comment (jim-minter) [inline pkg/operator/controllers/workaround/systemreserved.go:63]: nit: can write the following to simplify and remove the json.Marshal:

```
	kc := &mcv1.KubeletConfig{
		...
		Spec: mcv1.KubeletConfigSpec{
			KubeletConfig: &runtime.RawExtension{
				Object: &unstructured.Unstructured{
					Object: map[string]interface{}{
						"systemReserved": map[string]interface{}{
							"memory": memReserved,
						},
						"evictionHard": map[string]interface{}{
							"memory.available": hardEviction,
						},
					},
				},
			},
		},
	}
```

**Review by jim-minter — APPROVED** [review]

**Comment by jim-minter** [comment]
lgtm

**Pull request merged**